### PR TITLE
[DMS-265] Kubernetes setup for official Docker image

### PR DESCRIPTION
**Pull request by msilesgap**
Two Folders were created 
```
eng/kubernetes/development  --> will use the existing local image
eng/kubernetes/production --> will use the official image from DockerHub
```

Files were duplicated for now, during the research I found this [Kustomize](https://blog.stack-labs.com/code/kustomize-101/) that could help to use some templates for the config files. 

From the other hand  the image can be updated using this command `kubectl set image deployment/my-deployment mycontainer=myimage` 

`kubectl set image deployment/data-management-service data-management-service=edfialliance/data-management-service:pre`

### CODE REVIEW

File: eng/kubernetes/development/data-management-service-deployment.yaml [hunk 1]
```diff
@@ -18,7 +18,6 @@ spec:
       containers:
         - name: data-management-service
           image: local/edfi-data-management-service:latest
-          imagePullPolicy: Never
```
Comment (stephenfuqua) [inline eng/kubernetes/development/data-management-service-deployment.yaml:21]: Andony seemed to think this line was necessary. Are you sure we can delete it?

**Comment by stephenfuqua** [comment]
Kustomize sounds interesting. Helm is used by some of our customers, but I don't see a need for us to learn it at this time. I'll create a spike ticket for evaluating Kustomize in more detail at another time.

File: eng/kubernetes/development/data-management-service-deployment.yaml (hunk 1 above)
Comment (msilesgap) [reply eng/kubernetes/development/data-management-service-deployment.yaml]: It was my bad, I removed it when I was testing the image pull from docker hub.

**Comment by CSR2017** [comment]
@stephenfuqua
Can we consider using a PowerShell script to update the image name in the .yaml file, or should we duplicate just the data-management-service-deployment.yaml file and then use a PowerShell script to choose the list of files to apply with kubectl apply command ?

**Comment by stephenfuqua** [comment]
> @stephenfuqua Can we consider using a PowerShell script to update the image name in the .yaml file, or should we duplicate just the data-management-service-deployment.yaml file and then use a PowerShell script to choose the list of files to apply with kubectl apply command ?

There are a lot of ways to solve this. For right now, I just wanted a simple setup that makes it easy to run either way. I prefer to avoid the PowerShell solution you describe, as I want to keep this easy for any Linux or MacOS X users who don't have PowerShell.

File: eng/kubernetes/README.md [hunk 2]
```diff
@@ -137,3 +138,12 @@ Docker CLI context "default"`:
...
 4. Delete the minikube instance and start over.
+
+### Pods image status
+
+If you are facing the following errors with the Pods Status `kubectl get pods`
+
+`ErrImagePull` or `ImagePullBackOff`
+
+1. Review the status of the image, if you are using the local version make sure the image already exist; and if you are using the production configuration make sure the `image name` is valid. 
+2. Another useful command to get more details about the Pod Status `kubectl describe <POD_NAME>`
```
[the remark this answers is not in the document]
Comment (msilesgap) [reply eng/kubernetes/README.md:153]: Done

**Review by stephenfuqua — APPROVED** [review]

**Comment by stephenfuqua** [comment]
Actions aren't triggering after updating from `main`. I will close and re-open.

**Pull request merged**